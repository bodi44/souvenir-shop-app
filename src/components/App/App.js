import React from 'react'
import PropTypes from 'prop-types'

import './App.scss'
import BEM from '../../helpers/BEM'

const b = BEM('App')

const App = props => (
  <div className={b()}>
    <h1 className={b('title')}>Souvenir-store App</h1>
    {props.message ? <div className={b('data-error')}>{props.message}</div> : props.children}
  </div>
)

App.propTypes = {
  message: PropTypes.string,
  children: PropTypes.node
}

export default App