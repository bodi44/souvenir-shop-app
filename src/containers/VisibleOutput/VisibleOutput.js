import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import {
  getAllPurchasesByDate,
  getTotalIncome,
  isStoreHasErrorMessage,
  showHelpMessage
} from '../../reducers'
import Output from '../../components/Output/'
import HelpMessage from '../../components/HelpMessage/HelpMessage'

const VisibleOutput = ({ allPurchases, totalIncome, storeErrorMessage, helpMessage }) => {
  if (storeErrorMessage) {
    return (
      <Output title={'Error:'} error={'error'}>
        {storeErrorMessage} <br/> To get instructions please type: <strong>help</strong>
      </Output>
    )
  }

  if (totalIncome) {
    return (
      <Output title={'Output:'} output={totalIncome}/>
    )
  }

  if (helpMessage) {
    return (
      <Output title={'Help:'}>
        <HelpMessage/>
      </Output>
    )
  }

  return (
    <Output title={'Output:'} allPurchases={allPurchases}/>
  )
}

Output.propTypes = {
  allPurchases: PropTypes.array.isRequired,
  totalIncome: PropTypes.string,
  storeErrorMessage: PropTypes.string,
  helpMessage: PropTypes.bool
}

export default connect(
  state => ({
    allPurchases: getAllPurchasesByDate(state),
    totalIncome: getTotalIncome(state),
    storeErrorMessage: isStoreHasErrorMessage(state),
    helpMessage: showHelpMessage(state)
  }),
  null
)(VisibleOutput)
