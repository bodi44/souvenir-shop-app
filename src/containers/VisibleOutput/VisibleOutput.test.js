import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import expect from 'expect'
import Adapter from 'enzyme-adapter-react-16'

import App from '../../components/App'
import VisibleOutput from './VisibleOutput'

Enzyme.configure({ adapter: new Adapter() })

describe('VisibleOutput component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <App>
        <VisibleOutput/>
      </App>
    )

    expect(component).toMatchSnapshot()
  })
})