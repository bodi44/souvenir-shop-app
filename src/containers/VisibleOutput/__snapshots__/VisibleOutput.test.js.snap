// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibleOutput component should render correctly 1`] = `ShallowWrapper {}`;
